### PR TITLE
Better group plan member counts

### PR DESCRIPTION
**Pull request by SabreCat**
[//]: # (Note: See http://habitica.wikia.com/wiki/Using_Your_Local_Install_to_Modify_Habitica%27s_Website_and_API for more info)

[//]: # (Put Issue # here, if applicable. This will automatically close the issue if your PR is merged in)
Further fixage for #10124 .

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)

1. Adds a migration to reconcile `memberCount` with data from the user collection for all existing Group Plans.
2. Adds a step at time of group plan creation to reconcile the group's `memberCount` before setting up subscriptions.
3. Adds a server config variable for the connection string used by migrations. This should be more secure than editing connection strings directly in migration files.
4. Cleans up an old migration that had a garbage connection string.

### To Do

* [x] Run some local tests / dry runs against beta server

[//]: # (Put User ID in here - found on the Habitica website at User Icon > Settings > API)

### CODE REVIEW

**Comment by TheHollidayInn** [comment]
Woot!

File: migrations/groups/reconcile-group-plan-members.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+          ]
+      }
+    );
+    if (group.memberCount !== canonicalMemberCount) {
+      pause();
+      return dbGroups.update(
+        {_id: group._id},
+        {$set: {memberCount: canonicalMemberCount}}
+      ).then(() => {
+        await stripePayments.chargeForAdditionalGroupMember(group);
```
Comment (TheHollidayInn) [inline migrations/groups/reconcile-group-plan-members.js]: We need to check to ensure this is a group plan with Stripe. Amazon is also an option, but doesn't need a charge like this.

**Comment by Alys** [comment]
Should this also fix `quantity` and use that as another check for whether Stripe needs to be updated?

**Comment by SabreCat** [comment]
@Alys It does fix `quantity` via this line https://github.com/HabitRPG/habitica/blob/04d7ff13def23a4265075a903f586eb18684a099/website/server/libs/payments/stripe.js#L160, but I can definitely check to see if there are any discrepancies in `quantity` that don't correspond to discrepancies in `memberCount`!

**Pull request merged**